components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManager.java

```diff
@@ -60,6 +60,18 @@ BrandingPreference getBrandingPreference(String type, String name, String locale
     BrandingPreference resolveBrandingPreference(String type, String name, String locale)
             throws BrandingPreferenceMgtException;
 
+    /**
+     * This API is used to retrieve a resolved branding preference for an application.
+     *
+     * @param type   Type of the branding preference.
+     * @param name   Name of the application.
+     * @param locale Language preference of the branding.
+     * @return The resolved branding preference. If not exists return the default branding preference.
+     * @throws BrandingPreferenceMgtException if any error occurred.
+     */
+    BrandingPreference resolveApplicationBrandingPreference(String type, String name, String locale)
```

Comment:
What are the possible values for `type`?
Is that type indicate org level branding/application level branding?
The method name implies that this method is for application branding preference. So do we need type param still?

Reply:
Addressed with https://github.com/wso2-extensions/identity-branding-preference-management/pull/13/commits/b823151d847b7de7891b003b16a1e244ef6c3329